category/fold.lean

```diff
@@ -0,0 +1,237 @@
+/-
+Copyright (c) 2018 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Simon Hudon, Sean Leather
+
+List folds generalized to `traversable`.
+-/
+
+import data.list.basic
+import category.traversable.instances category.traversable.lemmas
+import category_theory.category category_theory.types category_theory.opposites category_theory.examples.monad
+import category.applicative
+
+universes u v
+
+open ulift category_theory
+
+namespace functor
+
+section monoids
```

Comment:
Perhaps this `section` is unnecessary. Instead, can we get a signposting comment, something like ``/- Applicative functors used internally by the implementation of `traversable` folds. -/``.